server/src/main/java/org/elasticsearch/action/admin/indices/alias/Alias.java

```diff
@@ -223,6 +223,15 @@ public static Alias fromXContent(XContentParser parser) throws IOException {
         while ((token = parser.nextToken()) != XContentParser.Token.END_OBJECT) {
             if (token == XContentParser.Token.FIELD_NAME) {
                 currentFieldName = parser.currentName();
+                // check if there are any unknown fields
+                if (currentFieldName.equals(FILTER.getPreferredName()) == false &&
```

Comment:
Maybe extract all known names to set once and then just check it with `knownFields.contains(currentFieldName) == false`

Reply:
OK, I have changed the code yet.